app/uk/gov/hmrc/gform/obligation/ObligationService.scala

```diff
@@ -84,10 +84,27 @@ class ObligationService(gformConnector: GformConnector) {
     for {
       idNumbers <- hmrcTaxPeriodIdentifiers.nonEmptyTraverse(i =>
                     withEvaluatedIdB(formTemplate, authService, retrievals, form, i))
-      taxResponses <- gformConnector.getAllTaxPeriods(idNumbers)
-      obligations = updatedObligations(idNumbers, taxResponses)
+      filteredIdNumbers <- Future(idNumbers.filter(i => i.idNumberValue.value != ""))
+      obligations       <- checkIfIdNumbersExist(filteredIdNumbers, idNumbers, form)
     } yield form.copy(obligations = RetrievedObligations(obligations))
 
+  def checkIfIdNumbersExist(
+    filteredIdNumbers: List[HmrcTaxPeriodWithEvaluatedId],
+    idNumbers: NonEmptyList[HmrcTaxPeriodWithEvaluatedId],
+    form: Form)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[List[TaxPeriodInformation]] =
+    filteredIdNumbers match {
+      case x :: xs => {
+        for {
+          taxResponses <- gformConnector.getAllTaxPeriods(NonEmptyList(x, xs))
```

Comment:
just do a map instead of a for comprehension, there is only 1 call here.

Reply:
case x::xs => gformConnector.getAllTaxPeriods(NonEmptyList(x, xs)).map{
......
}